streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -194,11 +199,12 @@ public void computeTaskLags(final Map<TaskId, Long> allTaskEndOffsetSums) {
             final Long offsetSum = taskOffsetSums.getOrDefault(task, 0L);
 
             if (endOffsetSum < offsetSum) {
-                throw new IllegalStateException("Task " + task + " had endOffsetSum=" + endOffsetSum +
-                                                    " smaller than offsetSum=" + offsetSum);
-            }
-
-            if (offsetSum == Task.LATEST_OFFSET) {
+                LOG.warn("Task " + task + " had endOffsetSum=" + endOffsetSum +
+                             " smaller than offsetSum=" + offsetSum + ". This probably means the task is corrupted," +
+                             " which in turn indicates that it will need to restore from scratch, so we pin the lag" +
+                             " to the end offset of the log.");
+                taskLagTotals.put(task, endOffsetSum);
```

Comment:
prop: Could you explain a bit better what the warning is about? If somebody does not know the code, it is hard to understand what is going on.